stackslib/src/net/download/nakamoto/mod.rs

```diff
@@ -176,6 +176,14 @@ pub use crate::net::download::nakamoto::tenure_downloader_unconfirmed::{
     NakamotoUnconfirmedDownloadState, NakamotoUnconfirmedTenureDownloader,
 };
 
+pub fn downloader_block_height_to_reward_cycle(
+    pox_constants: &PoxConstants,
+    first_block_height: u64,
+    block_height: u64,
+) -> Option<u64> {
+    pox_constants.block_height_to_reward_cycle(first_block_height, block_height.saturating_sub(1))
```

Comment:
Why does the downloader_block_height_to_reward_cycle always call block_height - 1?

Reply:
The downloader counts reward cycles in terms of PoX reward sets, not signer sets.  Signer sets take effect at reward index 0, but PoX reward sets take effect at reward index 1 (since the first PoX payouts happen at the 1st, not 0th, block in the reward cycle).

This behavior is needed because the Nakamoto chains coordinator processes all blocks for which it can determine the PoX recipients (i.e. it also counts reward cycles by PoX rewards sets, and necessarily so).  The downloader is tied to the chains coordinator's behavior because it needs to process _exactly_ the tenures that the coordinator's sortition history describes in order to reliably download the anchor block for the next cycle (thereby unblocking the chains coordinator).